### DISCUSSION TITLE
Failed to connect: libaws-c-mqtt: AWS_ERROR_MQTT_UNEXPECTED_HANGUP, The connection was closed unexpectedly

### DISCUSSION THREAD

**Discussion by hchautrung** [aws/aws-iot-device-sdk-js-v2#479] [Q&A] [answered]
Hi everyone,

I follow the instruction on the main page to run the sample in `samples/node/pub_sub` with the provided command  
`node dist/index.js --endpoint <endpoint> --cert <file> --key <file> --ca_file <file>`  

However, I got this error `Failed to connect: libaws-c-mqtt: AWS_ERROR_MQTT_UNEXPECTED_HANGUP, The connection was closed unexpectedly`. 

Note: Im sure that cert, key and ca files are all correct because I can connect and do pub/sub with [aws-iot-device-sdk-python-v2](https://github.com/aws/aws-iot-device-sdk-python-v2) or use `paho-mqtt` with python.

Kindly advise.
Thanks

**Comment by jmklix** [comment, marked as answer]
What does the IAM policy for the certificates look like?

You can always go to the AWS console and follow the IoT getting started guide. It will download new certs and a shell script to get the pubsub running. Trying that will help us determine what might be causing your UNEXPECTED_HANGUP
